File: ios/MyModule.m
#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_MODULE(MyModule, RCTEventEmitter)
RCT_EXTERN_METHOD(getName:(NSString *)name resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/MyModule.swift
import Foundation
import AVFoundation
import React

@objc(MyModule)
class MyModule: NSObject, RCTBridgeModule{
  static func moduleName() -> String! {
    return "MyModule";
  }
  
  @objc
  func getName(_ name: String?, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
    resolver("This is iOS MyModule \(name)")
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
      return true
  }
}
